datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -132,49 +148,45 @@ class except the :py:meth:`check` method but sometimes it might be useful for a
                 older versions of the Agent).
         """
-        self.metrics = defaultdict(list)
-        self.check_id = ''
-        self.instances = kwargs.get('instances', [])
-        self.name = kwargs.get('name', '')
-        self.init_config = kwargs.get('init_config', {})
-        self.agentConfig = kwargs.get('agentConfig', {})
-        self.warnings = []
+        name = kwargs.get('name', '')
+        init_config = kwargs.get('init_config', {})
+        agentConfig = kwargs.get('agentConfig', {})
+        instances = kwargs.get('instances', [])
 
         if len(args) > 0:
-            self.name = args[0]
+            name = args[0]
         if len(args) > 1:
-            self.init_config = args[1]
+            init_config = args[1]
         if len(args) > 2:
             # agent pass instances as tuple but in test we are usually using list, so we are testing for both
             if len(args) > 3 or not isinstance(args[2], (list, tuple)) or 'instances' in kwargs:
                 # old-style init: the 3rd argument is `agentConfig`
-                self.agentConfig = args[2]
+                agentConfig = args[2]
                 if len(args) > 3:
-                    self.instances = args[3]
+                    instances = args[3]
             else:
                 # new-style init: the 3rd argument is `instances`
-                self.instances = args[2]
+                instances = args[2]
 
         # NOTE: Agent 6+ should pass exactly one instance... But we are not abiding by that rule on our side
         # everywhere just yet. It's complicated... See: https://github.com/DataDog/integrations-core/pull/5573
-        self.instance = self.instances[0] if self.instances else None
+        instance = instances[0] if instances else None
+
+        self.check_id = ''
+        self.name = name  # type: str
+        self.init_config = init_config  # type: InitConfigType
+        self.agentConfig = agentConfig  # type: AgentConfigType
+        self.instance = instance  # type: Optional[InstanceType]
+        self.instances = instances  # type: List[InstanceType]
+        self.warnings = []  # type: List[str]
+        self.metrics = defaultdict(list)  # type: DefaultDict[str, List[str]]
 
         # `self.hostname` is deprecated, use `datadog_agent.get_hostname()` instead
-        self.hostname = datadog_agent.get_hostname()
+        self.hostname = datadog_agent.get_hostname()  # type: str
 
         logger = logging.getLogger('{}.{}'.format(__name__, self.name))
         self.log = CheckLoggingAdapter(logger, self)
 
-        # Provides logic to yield consistent network behavior based on user configuration.
-        # Only new checks or checks on Agent 6.13+ can and should use this for HTTP requests.
-        self._http = None
-
-        # Used for sending metadata via Go bindings
-        self._metadata_manager = None
-
-        # Save the dynamically detected integration version
-        self._check_version = None
```

Comment:
👏  documenting the methods with docstrings instead of comments here will be helpful.

Reply:
This isn't a request for change, btw, just a comment that the new approach is better!

Reply:
@mgarabed Yup, gotcha!